collection/collection_map.go

```diff
@@ -7,31 +7,60 @@ import (
 	"regexp"
 
 	"github.com/corazawaf/coraza/v3/internal/corazarules"
+	"github.com/corazawaf/coraza/v3/internal/strings"
 	"github.com/corazawaf/coraza/v3/types"
 	"github.com/corazawaf/coraza/v3/types/variables"
 )
 
+// anchoredVar stores the case preserved Original name and value
+// of the variable
+type anchoredVar struct {
+	// Name is the Key sensitive name
+	Name  string
+	Value string
+}
+
 // Map are used to store VARIABLE data
 // for transactions, this data structured is designed
 // to store slices of data for keys
 // Important: CollectionMaps ARE NOT concurrent safe
 type Map struct {
-	data     map[string][]types.AnchoredVar
-	name     string
-	variable variables.RuleVariable
+	data              map[string][]anchoredVar
+	name              string
+	variable          variables.RuleVariable
+	caseSensitiveKeys bool
```

Comment:
caseSensitiveKeys name signifies something like a set or list. Can we use a name that signifies a boolean?